### PR TITLE
nrf-command-line-tools: init at 10.16.0 , segger-jlink: init at 7.66

### PR DESCRIPTION
**Pull request by StarGate01**
###### Description of changes

This PR adds the Nordic Semiconductor NRF command-line tools (`nrf-command-line-tools`), providing tools like `nrfjprog`. It also updates `segger-jlink` , which is a runtime dependency of the NRF tools.

This PR supersedes #178603 (`segger-jlink`update), which continues the stale / abandoned #121601 . Based on the work by @FlorianFranzen and @reardencode . Note that the udev rules for `segger-jlink` have to be applied via `services.udev.packages`, and the user has to be member of `dialout`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -14663,4 +14663,10 @@
     github = "npatsakula";
     githubId = 23001619;
   };
+  stargate01 = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:14666]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/development/tools/misc/segger-jlink/default.nix [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  nativeBuildInputs = [ autoPatchelfHook ];
+  buildInputs = [ qt4 udev ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  installPhase = ''
+    # Install all files to JLink first
+    mkdir -p $out/JLink
+    cp -r * $out/JLink
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/segger-jlink/default.nix:84]: We shouldn't polute /JLink

File: pkgs/development/tools/misc/nrf-command-line-tools/default.nix [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  runtimeDependencies = [ segger-jlink ];
+
+  nativeBuildInputs = [ autoPatchelfHook ];
+  buildInputs = [ udev libusb ];
+
+  dontConfigure = true;
+  dontBuild = true;
+
+  outputs = [ "out" ];
+
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/nrf-command-line-tools/default.nix:59]:
```suggestion
```

thats the default

File: pkgs/development/tools/misc/nrf-command-line-tools/default.nix [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      sha256 = "ba208559ae1195a0d4342374a0eb79697d31d6b848d180ac906494f17f56623b";
+    };
+    aarch64-linux = {
+      name = "linux-arm64";
+      sha256 = "cffa4b8becdb5545705fd138422c648d809b520b7bc6c77b8b50aa1f79ebe845";
+    };
+    armv7l-linux = {
+      name = "linux-armhf";
+      sha256 = "c58d330152ae1ef588a5ee1d93777e18b341d4f6a2754642b0ddd41821050a3a";
+    };
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/nrf-command-line-tools/default.nix:28]: 32 bit OS's don't have good support in NixOS.

Comment (StarGate01) [reply pkgs/development/tools/misc/nrf-command-line-tools/default.nix:28]: Should I remove the support then?

File: pkgs/development/tools/misc/segger-jlink/default.nix (hunk 2 above)
Comment (StarGate01) [reply pkgs/development/tools/misc/segger-jlink/default.nix]: Okay, I unpacked the package properly.

File: pkgs/development/tools/misc/nrf-command-line-tools/default.nix [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+  };
+
+  platform = supported.${system} or (throw "unsupported platform ${system}");
+
+  versionSem = {
+    major = "10";
+    minor = "16";
+    patch = "0";
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/nrf-command-line-tools/default.nix:37]: Please use lib.versions.(major|minor|patch) instead.

**Comment by SuperSandro2000** [comment]
Also please squash the commits so that we have one for each package and one for your maintainer entry.

File: pkgs/development/tools/misc/nrf-command-line-tools/default.nix (hunk 4 above)
Comment (StarGate01) [reply pkgs/development/tools/misc/nrf-command-line-tools/default.nix:28]: I'll leave it in then.

**Comment by SuperSandro2000** [comment]
@ofborg build nrf-command-line-tools segger-jlink

File: pkgs/development/tools/misc/nrf-command-line-tools/default.nix [hunk 6]
```diff
@@ -0,0 +1,71 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, system
+, udev
+, libusb
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/nrf-command-line-tools/default.nix:7]:
```suggestion
, libusb1
```

File: pkgs/development/tools/misc/segger-jlink/default.nix [hunk 7]
```diff
@@ -0,0 +1,121 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, qt4
+, udev
+, system
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/segger-jlink/default.nix:7]:
```suggestion
```
This must be taken from stdenv.

File: pkgs/development/tools/misc/nrf-command-line-tools/default.nix [hunk 8]
```diff
@@ -0,0 +1,71 @@
+{ lib
+, stdenv
+, fetchurl
+, autoPatchelfHook
+, system
```
Comment (SuperSandro2000) [inline pkgs/development/tools/misc/nrf-command-line-tools/default.nix:5]:
```suggestion
```
This must be taken from stdenv.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please make sure that ``nix-build -A`` and the name of the package works.

**Comment by StarGate01** [comment]
> nix-build -A 

I have confirmed that loading and building the package via my configuration.nix works, however I am unsure on how to build the package specifically on the commandline using `nix-build`. I need to specify `nixpkgs.config.segger-jlink.acceptLicense = true`, but my tests using `--option` or `--attr` were unsuccessful. How do other packages with `acceptLicense` handle this?

**Comment by MatthewCroughan** [comment]
@StarGate01 This is how you would build it, if you're checked out on your branch of nixpkgs `NIXPKGS_ALLOW_UNFREE=1 nix-build -E '(import ./. {}).segger-jlink.override { acceptLicense = true; }'`

**Comment by StarGate01** [comment]
@MatthewCroughan thank you! That is really good to know.

**Review by MatthewCroughan — APPROVED** [review]

**Comment by ajs124** [comment]
Is there any version of this that doesn't depend on qt4, because we've been trying to get rid of that over in #174634

**Comment by StarGate01** [comment]
@ajs124 Since this is a closed-source binary package, no other option is available.

The only alternative would be to use the included Qt4 libraries instead of linking to the nixpkgs provided ones. That would require some patching of the included libraries though I assume.

**Comment by ajs124** [comment]
That's sad…

> The only alternative would be to use the included Qt4 libraries instead of linking to the nixpkgs provided ones. That would require some patching of the included libraries though I assume.

Can you look into that, maybe?

**Comment by MatthewCroughan** [comment]
I thought it would be trivial to backport. Guess not. Oh well!
